_posts/2020-04-07-trying-out-julia.md

```diff
@@ -0,0 +1,227 @@
+
+* Julia seems to be gaining some momentum in the programming and especially data science community.
+* Even as Julia is dynamically typed it has some static typing going on by enabling the coder to declare a type if he or she wants to. This is akin of Python's type annotation though it seems a little but more powerful. There is a long documentation about this at [docs.julialang.org/en/v1/manual/types/](https://docs.julialang.org/en/v1/manual/types/). One very positive thing I read there is that you can only sub type abstract types not concrete ones, as a Spring framework user this warms my mind a lot ;) !
+* The benchmarks seems to indicate it's much faster than the main competition Python. This is due the fact the language is compiled to native code using LLVM.
+* You can utilize Python packages.
+* I have some background with MatLab and similar plotting stuff seems to be the forte of Julia.
+
+Plotting is actually the "entry point" for this blog. I'll try to do same kind of stuff I did with MatLab with Julia. So basically this means scripting some plotting. So I won't delve into the finer points of e.g. building server side software.
+
+So if you want deep knowledge and insight into Julia as a language this is not the blog post for you, but if you just want to know how to display some data and setup your own Julia environment please do continue.
+
+## Okay how to install this thing
+
+Well let's google it. The page [julialang.org/](https://julialang.org/) is the home page and it has a big download button. Straight away the installation is a bit bare bones since it's just a tar-package (mind you I'm using Linux based system) and not some fancy installer or CLI-tool. After I've added the extracted bin folder to my PATH it's time to take the first sip with the repl.
+
+![Julia REPL](/img/trying-out-julia/julia-repl.png)
+
+Great we have a fancy calculator which seems to be even calculating correctly! After some 20-20 hindsight I can say there is some important information already displayed in the welcome ASCII-art, mainly the ]-mark. This is the way to access Julia package manager for adding dependencies. Also with hindsight using `activate <env_name>` with the package manager allows you to create python like virtual environments to contain your packages within your project.
+
+So good so far but I'm one of those guys who likes IDEs especially when learning new languages. The Julia homepage has a nice list of supported and familiar editors and IDEs. But why use something you already know? Let's go all in with new stuff and select Atom editor with JUNO plug-in. Ok yes part of the reason selecting Juno was that it looked cool in screenshots at [junolab.org](https://junolab.org). The installation was quite straight forward as Atom had a .deb-package and after installing that I just needed to add the uber-juno plug-in with atom's built in tools. Yes now I had all the tools in place, time to start coding.
```

Comment:
Sanonta lienee So far so good, but I'm.

Ok yes ainakin pilkku väliin. Queens english versio olisi jotain: I must confess, that the reason for selecting Juno...

straighforward kirjoitetaan mun mielestä yleensä yhteen.

installing **it**, I just...

Yes pois vikan lauseen alusta